services/QuillLMS/app/models/activity.rb

```diff
@@ -313,6 +313,18 @@ def serialize_with_topics_and_publication_date
     serializable_hash.merge({topics: topics&.map(&:genealogy), publication_date: publication_date})
   end
 
+  def update_questions_flag_status_if_necessary!
+    return unless flags.include?(:production)
+
+    question_ids = data["questions"].map{|q| q["key"]}
+    questions = Question.where("uid in (?)", question_ids)
```

Comment:
Tiny, but I'd rename this variable `question_uids` to avoid ambiguity (since questions do also have ids).